src/qforte/utils/qbt_tapering.py

```diff
@@ -0,0 +1,318 @@
+"""
+Functions for performing qubit tapering based on the Z2 symmetries of the qubit Hamiltonian
+"""
+
+import qforte as qf
+import numpy as np
+from itertools import product
+
+def find_Z2_symmetries(hamiltonian, qiskit_order = False):
+
+    """
+    This function computes the Z2 symmetries of the Hamiltonian, identifies the ID numbers of the qubits to
+    be tapered off, provides the corresponding sign structures, and constructs the necessary unitary operators.
+    It is based on the qubit tapering approach reported in https://arxiv.org/abs/1701.08213.
+
+    Arguments:
+
+    hamiltonian: QubitOperator
+        The qubit Hamiltonian whose Z2 symmetries we want to find.
+
+    qiskit_order: bool
+        If True, the qubits to be tapered off will have the smallest id numbers possible, as is done in Qiskit.
+        If False, the qubits to be tapered off will have the largest id numbers possible. This is the default,
+        since it leads to less operations once the qubits are removed.
+    """
+
+    # The algorithm consists of the following steps:
+    # 1) Convert qubit Hamiltonian to binary matrix, called the parity check matrix E
+    # 2) Find the basis of the kernel of the parity check matrix, ker{E}
+    # 3) Find maximal Abelian subgroup of the basis of ker{E}
+    # 4) Construct unitary matrices that will be used in the qubit_tapering algorithm
+    # 5) Compute the various signs corresponding to the eigenvalues of the sigma_x operators that will be removed
+
+    # 1) Convert qubit Hamiltonian to binary matrix
+
+    ## Obtain number of qubits and number of Pauli strings contained in qubit Hamiltonian
+    n_qubits = hamiltonian.num_qubits()
+    n_strings = len(hamiltonian.terms())
+
+    ## Construct parity check matrix
+    prt_chck_mtrx = np.zeros((n_strings, 2 * n_qubits), dtype=np.uint8)
```

Comment:
You should explicitly state the indexing conventions. Rows are Hamiltonian strings, which is simple enough. What is less straightforward is that qubits are indexed z_(n+1), ... z_0, x_(n+1), ..., x_0 rather than z_0, ..., z_(n+1), x_0, ..., x_(n+1)